records_mover/db/mysql/load_options.py

```diff
@@ -110,157 +112,339 @@ def generate_load_data_sql(self,
         return clause
 
 
-def mysql_load_options(unhandled_hints: Set[str],
-                       records_format: DelimitedRecordsFormat,
-                       fail_if_cant_handle_hint: bool) -> MySqlLoadOptions:
-    hints = records_format.validate(fail_if_cant_handle_hint=fail_if_cant_handle_hint)
+class MysqlLoadParameters:
+    """
+    A class representing MySQL load parameters.
 
-    #
-    # The server uses the character set indicated by the
-    # character_set_database system variable to interpret the
-    # information in the file. SET NAMES and the setting of
-    # character_set_client do not affect interpretation of input. If
-    # the contents of the input file use a character set that differs
-    # from the default, it is usually preferable to specify the
-    # character set of the file by using the CHARACTER SET clause. A
-    # character set of binary specifies “no conversion.”
-    #
-    hint_encoding: HintEncoding = hints.encoding
+    Attributes:
+        hints (ValidatedRecordsHints): Validated records hints.
+        hint_quoting (HintQuoting): Quoting hint.
+        fail_if_cant_handle_hint (bool): Fail if unable to handle hint.
+        records_format (DelimitedRecordsFormat): Records format.
+        unhandled_hints (Set[str]): Set of unhandled hints.
+    """
+    def __init__(self,
+                 fail_if_cant_handle_hint: bool,
+                 records_format: DelimitedRecordsFormat,
+                 unhandled_hints: Set[str]):
+        """Initialize the MySQL load parameters."""
+        self.hints: ValidatedRecordsHints = records_format.validate(
+            fail_if_cant_handle_hint=fail_if_cant_handle_hint)
+        self.hint_quoting: HintQuoting = self.hints.quoting
+        self.fail_if_cant_handle_hint: bool = fail_if_cant_handle_hint
+        self.records_format: DelimitedRecordsFormat = records_format
+        self.unhandled_hints: Set[str] = unhandled_hints
+
+
+def get_character_set(mysql_load_parameters: MysqlLoadParameters) -> MySqlCharacterSet:
+    """
+    Determine the character set for the given MySQL load parameters.
+    The server uses the character set indicated by the
+    character_set_database system variable to interpret the
+    information in the file. SET NAMES and the setting of
+    character_set_client do not affect interpretation of input. If
+    the contents of the input file use a character set that differs
+    from the default, it is usually preferable to specify the
+    character set of the file by using the CHARACTER SET clause. A
+    character set of binary specifies “no conversion.”
+
```

Comment:
you should also mention that on success, this removes 'encoding' from the load parameters unhandled_hints. its a little weird to do modifications in a "getter" but eh

Reply:
Very good point; I'm open to another name. Additionally, PEP conventions suggest to skip get_ names entirely.

Any ideas for prepends? Maybe it'd just be better to skip them.